scripts/sanitycheck

```diff
@@ -3099,7 +3099,9 @@ def chatty_test_cb(instances, goals, goal):
 
     info("{:>{}}/{} {:<25} {:<50} {} ({})".format(
         total_done, total_tests_width, total_tests, i.platform.name,
-        i.test.name, status, handler_type))
+        i.test.id, status, handler_type))
+    local_path = "/".join(i.test.name.split("/")[0:-1])
```

Comment:
This parsing is re-generating the `i.test.workdir` argument that was passed to (and lost by) the `TestCase(.., workdir,...)` constructor. I think it would be better to stop losing it there. It would also help to have all the relevant logic in that one place, for instance there's an ominous comment "workdir can be '.'" over there - have you tested this change with funky --testcase-root arguments and how does the output look like then?

Reply:
> have you tested this change with funky --testcase-root arguments

(copying `samples/hello_world/` somewhere else is a fast way to test this)

Reply:
FWIW, this works as intended with my case of a funky --testcase-root from #16920.